rt/rs/sse/src/main/java/org/apache/cxf/jaxrs/sse/OutboundSseEventImpl.java

```diff
@@ -92,6 +92,7 @@ public Builder data(GenericType newType, Object newData) {
 
         @Override
         public Builder data(Object newData) {
+            this.type = newData.getClass();
```

Comment:
Good catch!  Do you have any thoughts on what we should do if a null is passed in?  The spec throws an `IllegalArgumentException` if the user passes null to `sse.newEvent(String data)`.  For `builder.data(null)`, we could either do the same, or take null as the data and use Object.class as the default type.  I'm leaning toward throwing the exception.  What do you think?